### PR TITLE
Clean up protocol support in IPVS manager

### PR DESCRIPTION
**Pull request by drewkerrigan**
Addresses DCOS-13696.

### CODE REVIEW

**Comment by drewkerrigan** [comment]
Per @nlsun's suggestion, I will separate the UDP addiction from this PR.

File: src/minuteman_ipvs_mgr.erl [hunk 1]
```diff
@@ -280,23 +313,32 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
-handle_get_services(#state{netlink_generic = Pid, family = Family}) ->
-    AF = netlink_codec:family_to_int(inet),
-    Protocol = netlink_codec:protocol_to_int(tcp),
+handle_get_services(State) ->
+    lists:foldl(
+      fun(Protocol, Acc) ->
+              Acc ++ handle_get_services(inet, Protocol, State)
```
Comment (GoelDeepak) [inline src/minuteman_ipvs_mgr.erl]: Can we break this in two lines? For example Services = handle_get_services(inet, Protocol, State), [Services|Acc]. I am wondering why we used "++". Do we care about the order in the list?

File: src/minuteman_ipvs_mgr.erl [hunk 2]
```diff
@@ -280,23 +313,33 @@ code_change(_OldVsn, State, _Extra) ->
...
-    AF = netlink_codec:family_to_int(inet),
-    Protocol = netlink_codec:protocol_to_int(tcp),
+handle_get_services(State) ->
+    lists:foldl(
+      fun(Protocol, Acc) ->
+              Services = handle_get_services(inet, Protocol, State),
+              Acc ++ Services
+      end, [], ?IPVS_PROTOCOLS).
+
+handle_get_services(AddressFamily, Protocol, #state{netlink_generic = Pid, family = Family}) ->
```
Comment (aeyakovenko) [inline src/minuteman_ipvs_mgr.erl]: can you add a spec for this

File: src/minuteman_ipvs_mgr.erl [hunk 3]
```diff
@@ -280,23 +313,33 @@ code_change(_OldVsn, State, _Extra) ->
...
-    [proplists:get_value(service, MaybeService) || #netlink{msg = #new_service{request = MaybeService}} <- Replies,
+    [proplists:get_value(service, MaybeService)
+     || #netlink{msg = #new_service{request = MaybeService}}
+            <- Replies,
         proplists:is_defined(service, MaybeService)].
 
-handle_remove_service(IP, Port, State) ->
-    Protocol = netlink_codec:protocol_to_int(tcp),
-    Service = ip_to_address(IP) ++ [{port, Port}, {protocol, Protocol}],
+handle_remove_service(IP, Port, Protocol, State) ->
```
Comment (aeyakovenko) [inline src/minuteman_ipvs_mgr.erl]: can you add a spec for this too

File: src/minuteman_ipvs_mgr.erl [hunk 4]
```diff
@@ -305,10 +348,10 @@ handle_remove_service(Service, #state{netlink_generic = Pid, family = Family}) -
         _ -> error
     end.
 
-handle_add_service(IP, Port, #state{netlink_generic = Pid, family = Family}) ->
+handle_add_service(IP, Port, Protocol, #state{netlink_generic = Pid, family = Family}) ->
```
Comment (aeyakovenko) [inline src/minuteman_ipvs_mgr.erl]: spec...

**Comment by drewkerrigan** [comment]
@aeyakovenko added specs, and fixed a couple at the top as well.

**Pull request merged**